server/base/src/main/java/org/apache/accumulo/server/conf/store/PropStoreKey.java

```diff
@@ -89,25 +89,44 @@ protected PropStoreKey(final InstanceId instanceId, final String path, final ID_
    */
   public static @Nullable PropStoreKey<?> fromPath(final String path) {
     String[] tokens = path.split("/");
+    if (tokens.length < 1) {
+      return null;
+    }
 
     InstanceId instanceId;
     try {
       instanceId = InstanceId.of(tokens[IID_TOKEN_POSITION]);
+
+      // needs to start with /accumulo/[instanceId]
+      if (!path.startsWith(ZooUtil.getRoot(instanceId))) {
+        log.warn(
+            "Path '{}' is invalid for a property cache key, expected to start with /accumulo/[instance_id]",
```

Comment:
Is there a reason not to put the actual instanceId in the warning message?

Reply:
addressed in b9ef24d03b